src/main/java/com/eighthlight/fabrial/http/file/FileHttpResponder.java

```diff
@@ -182,24 +183,48 @@ private Response buildOptionsResponse(Request request, ResponseBuilder builder)
   }
 
   private Response buildGetDirectoryResponse(Request request, ResponseBuilder builder) {
-    var contents =
+    var list =
         fileController.getDirectoryContents(request.uri.getPath())
                       .stream()
                       .sorted()
-                      .reduce((p1, p2) -> p1 + "," + p2)
-                      .orElse("");
-
-    if (contents.isEmpty()) {
+                      .map(p -> {
+                        // would be nice to append "/" to directories
+                        // dir items must not contain leading slash
+                        // or the <base> element won't work
```

Comment:
> Thinking about this for a bit, I wonder if this limitation could imply the <base> element isn't quite the right fit for this use case.

I don't know enough about that element to say whether or not it's the "right" way to use it, but it seems to have the desired effect in my case. In any event, it shouldn't be too hard to just use the full path for reach file.

Reply:
> Accepting trailing / in requests makes sense to me

Sounds like there's a misunderstanding here. My comment from before about cob_spec is that it doesn't allow **leading** slashes in the link's `href` attribute. Having trailing slashes for directories is just a throwaway comment on how I'd prefer the "UI" to look (to distinguish files & directories).

Reply:
Oh no, I thought I did understand! Let me try again, and let me know if I'm still off (and maybe we can jump on a call if so).

I did understand [I think] that there are 2 separate question: prepending a [leading] slash and appending a [trailing] slash, and I was trying to address the trailing slash in the last paragraph of my comment above. Maybe that hurt more than it helped?

On the leading slash, I saw this in the PR description:

> The "Directory Links" cob_spec test fails—but only because it searches for href == /filename and my server uses href == filename.

and I assumed that meant "something different needs to happen to pass cob_spec" which led me to suggest using absolute paths. But in your last comment here you're saying

> My comment from before about cob_spec is that it doesn't allow leading slashes in the link's href attribute

which doesn't sound right to me, because it seems to contradict the PR description quote above, and also the [cob_spec selector that checks the links](https://github.com/8thlight/cob_spec/blob/db6a99ad3f29768f2184ac6c350adfa0e722d07c/src/main/java/HttpBrowser.java#L154). 

I'm thinking that maybe the `href == /filename` pseudocode might be abstract enough that it's causing my misunderstanding of what you mean, but in any case hopefully this comment makes more sense! 😄

Reply:
> But in your last comment here you're saying

Ack, that was a mistake. Other way around: cob_spec expects leading slashes, but I can't use them since it breaks the "relative path" behavior I get with `<base>`.

So you were right. I didn't fully grok your comment about updating cob_spec to accept trailing slashes, which I then mistakenly thought was somehow referencing my main issue with leading slashes causing cob_spec failures...

Ultimately, this is just another reason why I shouldn't leave "feature requests" for cob_spec in the comments of my server 😓 ...